### DISCUSSION TITLE
Do we need to account for concurrency?

### DISCUSSION THREAD

**Discussion by livelykitten** [ANLAB-KAIST/KENSv3#141] [Project Q&A 2. Transport Layer (TCP)] [answered]
There are four functions that we are going to be using:
1. systemCallback
2. packetArrived
3. timerCallback
4. sendPacket

Will they be running asynchronously? Do we have to account for race conditions?
e.g. another systemCallback starts running before existing systemCallback returns
e.g. packetArrived is invoked when the existing systemCallback has not returned

Thank you

**Comment by Pusnow** [comment, marked as answer]
No, you don't need to consider race conditions. All events (packet arrivals, system calls, and timers) arrive in a serialized manner.
